sdk/core/core-rest-pipeline/documentation/pipeline.md

```diff
@@ -0,0 +1,220 @@
+
+A `Pipeline` composes several policies and provides the ability to make a request that will flow through the policies in a specific order determined by their requirements. Which allows manipulating each request before and after it is made to the server.
+
+```typescript
+export interface AddPipelineOptions {
+  afterPhase?: PipelinePhase;
+  afterPolicies?: string[];
+  beforePolicies?: string[];
+  phase?: PipelinePhase;
+}
+
+export interface Pipeline {
+  addPolicy(policy: PipelinePolicy, options?: AddPipelineOptions): void;
+  clone(): Pipeline;
+  getOrderedPolicies(): PipelinePolicy[];
+  removePolicy(options: { name?: string; phase?: PipelinePhase }): PipelinePolicy[];
+  sendRequest(httpClient: HttpClient, request: PipelineRequest): Promise<PipelineResponse>;
+}
+```
+
+Dependencies expressed when calling `addPolicy` are not sensitive to order. The following code snippets have identical effects:
+
+```typescript
+pipeline.addPolicy(barPolicy);
+pipeline.addPolicy(fooPolicy, { afterPolicy: "barPolicy" });
+
+// The below is the same as the above.
+// Policies don't have to be part of the pipeline in order to be
+// referenced by before/afterPolicy.
+pipeline.addPolicy(fooPolicy, { afterPolicy: "barPolicy" });
+pipeline.addPolicy(barPolicy);
+```
+
+Phases are predefined buckets of activity that pre-defined policies inside core will fall into. They are currently a fixed set and not customizable by pipeline consumers.
+
+```typescript
+export type PipelinePhase = "Deserialize" | "Serialize" | "Retry" | "Sign";
+```
+
+# Default Pipeline
+
+`@azure/core-rest-pipeline` exposes `createPipelineFromOptions` which creates pipeline with a predefined set of policies. The resulting pipeline can be configured through `PipelineOptions`
+
+```typescript
+/**
+ * Defines options that are used to configure the HTTP pipeline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: PipelineRetryOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxySettings;
+
+  /**
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: RedirectPolicyOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options - Options to configure a custom pipeline.
+ */
+declare function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline;
+```
+
+### Policies included in a default pipeline
+
+- proxyPolicy **(NODE Only)**
```

Comment:
I wonder if we could get an image badge of the Node logo for this (with alt text of course)

Reply:
Something like this?
![image](https://user-images.githubusercontent.com/21109913/154575764-2c06ed85-7884-41a9-8cd8-6048d3d6565a.png)